### DISCUSSION TITLE
How can I make an index page of event-sourced objects by querying the aggregate roots?

### DISCUSSION THREAD

**Discussion by ericgross** [RailsEventStore/rails_event_store#1699] [General]
I would like to ask my aggregate root what the state is of my objects so that I can sort by this value in an index that shows millions of objects.

How can I get the current state of all of the objects in my system, each with their own events, without directly looking at the events, but rather looking at the aggregates?

**Comment by mpraglowski** [comment]
The best way will be to build a read model for that. Please check the https://blog.arkency.com/why-event-sourcing-basically-requires-cqrs-and-read-models/ post and other posts in this topic https://blog.arkency.com/tags/read-model/
